zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -561,7 +561,11 @@ public SessionTracker getSessionTracker() {
     }
 
     long getNextZxid() {
-        return hzxid.incrementAndGet();
+        return hzxid.get() + 1;
```

Comment:
Won't be a concern as long as we keep processing requests in order inside PrepRequestProcessor. If that changes this can break but that is not likely to happen.